File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] visited = new boolean[n];
        for(int i = 0 ; i < n ; i++){
            if(!visited[i]){
                dfs(visited, i, computers);
                answer++;
            }
        }     
        
        return answer;
    }
    void dfs(boolean[] visited, int idx,int[][] arr) {
        int[] cur = arr[idx];
        for(int i = 0 ; i < arr.length ; i++){
            if(i != idx && !visited[i] && cur[i] == 1){
                visited[i] = true;
                dfs(visited,i,arr);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(n, computers, start, visited):
    visited[start] = True
    for end in range(n):
        if end != start and computers[start][end] == 1:
            if visited[end] == False:
                dfs(n, computers, end, visited)

def solution(n, computers):
    visited = [False] * n
    answer = 0
    for i in range(n):
        if visited[i] == False:
            dfs(n, computers, i, visited)
            answer += 1
    return answer